File: frontend/src/types/document.ts
export interface PageSize {
  width: number;
  height: number;
}

export type PageLiteracyMode = "native_text" | "scanned" | "hybrid";

export interface DocumentMetadata {
  id: string;
  original_filename: string;
  page_count: number;
  page_sizes: PageSize[];
  render_scale: number;
  created_at: number;
}

export interface PageAnalysis {
  page_number: number;
  mode: PageLiteracyMode;
  has_meaningful_text: boolean;
  text_word_count: number;
  text_character_count: number;
  image_count: number;
  image_area_ratio: number;
}

export interface PageCropBox {
  x: number;
  y: number;
  w: number;
  h: number;
}

export interface PageInfo {
  page_number: number;
  width: number;
  height: number;
  preview_path: string | null;
  render_path: string | null;
  has_ocr: boolean;
  analysis: PageAnalysis | null;
  crop_box: PageCropBox | null;
}

File: backend/app/models/document.py
from pydantic import BaseModel
from typing import Literal, Optional


PageLiteracyMode = Literal["native_text", "scanned", "hybrid"]


class DocumentMetadata(BaseModel):
    id: str
    original_filename: str
    page_count: int
    page_sizes: list[dict]  # [{width, height}]
    render_scale: float
    created_at: float


class PageAnalysis(BaseModel):
    page_number: int
    mode: PageLiteracyMode
    has_meaningful_text: bool
    text_word_count: int
    text_character_count: int
    image_count: int
    image_area_ratio: float


class PageCropBox(BaseModel):
    x: float
    y: float
    w: float
    h: float


class PageInfo(BaseModel):
    page_number: int
    width: float
    height: float
    preview_path: Optional[str] = None
    render_path: Optional[str] = None
    has_ocr: bool = False
    analysis: Optional[PageAnalysis] = None
    crop_box: Optional[PageCropBox] = None
